### PR TITLE
Continously release versions.

### PR DESCRIPTION
**Pull request by jplaisted**
This is done via a custom GitHub workflow / action that uses shipkit github release and auto version plugins. This means we will publish a release on GitHub and bintray for every commit on a branch named `release/*`.

For now autopublish to bintray is still disabled, we can enable it once we're sure we're happy with this pipeline.

#4 
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by jplaisted** [comment]
Also, I acknowledge we should have some discussion here as to how we want to do development / releases.

Right now, this based on mockito. There is no "master" branch in this model. Instead, there are release branches (e.g. `release/0.1.x`). We can them make the highest version the default branch and work off that. We can keep old release branches around to port fixes back, if needed, or clean them up if we're no longer maintaining them.

File: docs/how/releases.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Releases
+
+We create [GitHub releases](https://github.com/linkedin/datahub-gma/releases) and publish our artifacts to
+[Bintray](https://bintray.com/linkedin/maven/datahub-gma).
+
+We use shipkit to help create our releases and auto-increment our versions.
```
Comment (mars-lan) [inline docs/how/releases.md:6]: Link for shipkit?

File: docs/how/releases.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+We automatically create releases for every commit on a branch that starts with `release/%semver%`, where semver is the
+wildcard semantic version of that branch (e.g. `release/1.0.x`).
+
+## Versioning
+
+As stated above, we use shipkit for automatically creating version tags. See
+[shipkit-auto-version for details](https://github.com/shipkit/shipkit-auto-version).
+
+We follow the [semantic versioning 2.0.0 convention](https://semver.org/). When bumping the major or minor version,
+please make a new `release` branch for it.
```
Comment (mars-lan) [inline docs/how/releases.md:17]: Understandable for major version, but creating a new branch for minor versions seems excessive?

File: docs/how/releases.md (hunk 1 above)
Comment (jplaisted) [reply docs/how/releases.md:6]: We aren't using the shipkit plugin itself but these smaller ones that I've linked below. Clarified.

Comment (jplaisted) [reply docs/how/releases.md:6]: (there is a "shipkit" plugin)

File: docs/how/releases.md (hunk 2 above)
Comment (jplaisted) [reply docs/how/releases.md:17]: /shrug

Up to you. I think it is important to remind ourselves that the minor version exists and we should be using it when we need to. Which is hopefully rare, but could happen. So I think in most cases we won't bump the minor anyway, so it is a bit moot. But when it does come up, might be nice to have?

Comment (jplaisted) [reply docs/how/releases.md:19]: Also while we're on `0` the minor version has more meaning and will actually be bumped more often.

Comment (jplaisted) [reply docs/how/releases.md]: Switched to trunk development. PTAL.

File: build.gradle [hunk 3]
```diff
@@ -155,4 +141,51 @@ spotless {
     target '**/*.md'
     prettier().config([printWidth: 120, proseWrap: 'always'])
   }
+}
+
+task getBranchName() {
+  doLast {
+    new ByteArrayOutputStream().withStream { os ->
+      def result = exec {
+        commandLine 'git', 'branch', '--show-current'
```
Comment (mars-lan) [inline build.gradle:150]: Seems a bit hacky to me? Hard to believe there's no existing lib/plugin that does this?

File: docs/how/releases/continuous-releases.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Automatic Changelog
+
+We use shipkit changelog to create the release notes of each version. See
+[shipkit-changelog](https://github.com/shipkit/shipkit-changelog) for more details.
+
+## Automatic Publishing
+
+Automation is done with a [GitHub workflow](../../.github/workflows/gh-version.yml) that will create the GitHub release
+and then publish to bintray for every commit on a `release` branch.
```
Comment (mars-lan) [inline docs/how/releases/continuous-releases.md:32]: main branch?

File: docs/how/releases/versioning.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# GMA Versioning
```
Comment (mars-lan) [inline docs/how/releases/versioning.md:1]: Seems redundant info?

**Review by mars-lan — APPROVED** [review]
Great work

**Pull request merged**